src/config/game.json

```diff
@@ -1,12 +1,1246 @@
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":106,
+                "costs":{
+                    "metal":0,
+                    "crystal":800,
+                    "deuterium":400,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":107,
+                "costs":{
+                    "metal":0,
+                    "crystal":4000,
+                    "deuterium":2000,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":108,
+                "costs":{
+                    "metal":400,
+                    "crystal":0,
+                    "deuterium":600,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":109,
+                "costs":{
+                    "metal":2000,
+                    "crystal":4000,
+                    "deuterium":600,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":110,
+                "costs":{
+                    "metal":10000,
+                    "crystal":20000,
+                    "deuterium":6000,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":111,
+                "costs":{
+                    "metal":200,
+                    "crystal":100,
+                    "deuterium":0,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":112,
+                "costs":{
+                    "metal":1000,
+                    "crystal":300,
+                    "deuterium":100,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":113,
+                "costs":{
+                    "metal":2000,
+                    "crystal":4000,
+                    "deuterium":1000,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":114,
+                "costs":{
+                    "metal":240000,
+                    "crystal":400000,
+                    "deuterium":160000,
+                    "energy":0,
+                    "factor":2
+                }
+            },
+            {
+                "unitID":115,
+                "costs":{
+                    "metal":0,
+                    "crystal":0,
+                    "deuterium":0,
+                    "energy":300000,
+                    "factor":3
+                }
+            }
+        ],
+        "requirements":{
+            "5":[
```

Comment:
Do you consider unitIDs strings or numbers? JavaScript allows numbers as keys, but JSON does not. This will need some extra code to rectify. 
It could also make sense to inline requirements because of this. My thought was that this wouldn't be used much and then a cheap lookup could achieve reconstruction, but this also applies other attributes.
Now that I think of it optimizing for human editability makes the most sense, which makes inline the best choice, so everything for a unit is always at one place. If needed the lookup map can be constructed from that.

Reply:
Resolved with new commit